pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -124,27 +124,28 @@ const (
 // AllParamTypes can be used for ParamType validation.
 var AllParamTypes = []ParamType{ParamTypeString, ParamTypeArray, ParamTypeObject}
 
-// ArrayOrString is modeled after IntOrString in kubernetes/apimachinery:
+// ParamValues is modeled after IntOrString in kubernetes/apimachinery:
 
-// ArrayOrString is a type that can hold a single string or string array.
+// ParamValue is a type that can hold a single string or string array.
 // Used in JSON unmarshalling so that a single JSON field can accept
 // either an individual string or an array of strings.
-// TODO (@chuangw6): This struct will be renamed or be embedded in a new struct to take into
-// consideration the object case after the community reaches an agreement on it.
-type ArrayOrString struct {
-	Type      ParamType `json:"type"` // Represents the stored type of ArrayOrString.
+type ParamValue struct {
+	Type      ParamType `json:"type"` // Represents the stored type of ParamValues.
 	StringVal string    `json:"stringVal"`
 	// +listType=atomic
 	ArrayVal  []string          `json:"arrayVal"`
 	ObjectVal map[string]string `json:"objectVal"`
 }
 
+// ArrayOrString is deprecated, this is to keep backward compatibility
+type ArrayOrString = ParamValue
```

Comment:
Shall we update the respective docs in api-spec.md?

Reply:
yes, it should be updated